### PR TITLE
KEYCLOAK-12845 KEYCLOAK-13312 KEYCLOAK-13313 KEYCLOAK-13196 Update RHSSO templates

### PR DESCRIPTION
**Pull request by mhajas**
Thanks for submitting your Pull Request!

Please make sure your PR meets the following requirements:

- [ ] Pull Request title is properly formatted: `[CLOUD-XYA] Subject`
- [ ] Pull Request contains link to the JIRA issue
- [ ] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket
- [ ] Attached commits represent units of work and are properly formatted
- [ ] You have read and agreed to the Developer Certificate of Origin (DCO) (see `CONTRIBUTING.md`)
- [ ] Every commit contains `Signed-off-by: Your Name <yourname@example.com>` - use `git commit -s`

### CODE REVIEW

**Comment by mhajas** [comment]
This PR adds the following:
1. Updates current templates to contain RHSSO 7.4 instead of RHSSO CD
2. Changes in https://github.com/jboss-container-images/redhat-sso-7-openshift-image/pull/129 were done to cd versions of templates. In this PR the commits from the 129 Pr are rebased to 7.4 version of templates and docs are regenerated
3. Script for generating OpenJ9 templates was added and templates were generated

@hmlnarik @iankko Could you please have a look? I can split into more PRs if you prefer, however, we would need to merge the first commit and then send the others after.

File: gen_openj9_templates.sh [hunk 1]
```diff
@@ -0,0 +1,13 @@
+#!/usr/bin/env bash
```
Comment (hmlnarik) [inline gen_openj9_templates.sh]: Please put this into `templates` directory

File: gen_openj9_templates.sh [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+# Remove old openj9 templates
+rm templates/openj9/*
+
+for original_template in templates/*.json; do
+  echo Creating OpenJ9 template for file "$original_template"
+  sed -e 's/OpenJDK/OpenJ9/' \
+    -e 's/sso74/sso74-openj9/' \
+    "$original_template" \
+    | tee "$(echo "$original_template" | sed -e 's_templates_templates/openj9_' -e 's/\([a-zA-Z0-9]*\)-/\1-openj9-/')" >/dev/null
```
Comment (hmlnarik) [inline gen_openj9_templates.sh]:
```suggestion
    > "$NEW_TEMPLATE"
```

File: gen_openj9_templates.sh [hunk 3]
```diff
@@ -0,0 +1,13 @@
+#!/usr/bin/env bash
+
+
+# Remove old openj9 templates
+rm templates/openj9/*
+
+for original_template in templates/*.json; do
+  echo Creating OpenJ9 template for file "$original_template"
+  sed -e 's/OpenJDK/OpenJ9/' \
+    -e 's/sso74/sso74-openj9/' \
```
Comment (hmlnarik) [inline gen_openj9_templates.sh]:
```suggestion
    -e 's/sso\([0-9]*\)/sso\1-openj9/' \
```

File: templates/openj9/sso74-openj9-image-stream.json [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+            },
+            "labels": {
+                "rhsso": "7.4.0.GA"
+            },
+            "spec": {
+                "tags": [
+                    {
+                        "name": "latest",
+                        "annotations": {
+                            "description": "Latest build of Red Hat Single Sign-On 7.4 OpenJ9 image",
```
Comment (hmlnarik) [inline templates/openj9/sso74-openj9-image-stream.json]:
```suggestion
                            "description": "Latest build of Red Hat Single Sign-On 7.4 on OpenJ9 image",
```

File: templates/openj9/sso74-openj9-image-stream.json [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+                            "supports": "sso:7.4",
+                            "version": "1.0",
+                            "openshift.io/display-name": "Red Hat Single Sign-On 7.4 on OpenJ9"
+                        },
+                        "referencePolicy": {
+                            "type": "Local"
+                        },
+                        "from": {
+                            "kind": "DockerImage",
+                            "name": "registry.redhat.io/redhat-sso-7/sso73-openshift:1.0"
```
Comment (hmlnarik) [inline templates/openj9/sso74-openj9-image-stream.json]: sso73??

File: templates/openj9/sso74-openj9-x509-mysql-persistent.json [hunk 6]
```diff
@@ -0,0 +1,652 @@
...
+    "kind": "Template",
+    "apiVersion": "v1",
+    "metadata": {
+        "annotations": {
+            "iconClass" : "icon-sso",
+            "tags" : "sso,keycloak,jboss",
+            "version": "7.4.0.GA",
+            "openshift.io/display-name": "Red Hat Single Sign-On 7.4 on OpenJ9 + MySQL (Persistent)",
+            "openshift.io/provider-display-name": "Red Hat, Inc.",
+            "description": "An example application based on RH-SSO 7.4 OpenJ9 image. For more information about using this template, see https://github.com/jboss-container-images/redhat-sso-7-openshift-image/docs.",
```
Comment (hmlnarik) [inline templates/openj9/sso74-openj9-x509-mysql-persistent.json]:
```suggestion
            "description": "An example application based on RH-SSO 7.4 on OpenJ9 image. For more information about using this template, see https://github.com/jboss-container-images/redhat-sso-7-openshift-image/docs.",
```

**Review by hmlnarik — CHANGES_REQUESTED** [review]
Thanks @mhajas 
Please see inline. Also update to `docs` directory to reflect the changes as well

File: docs/README.adoc [hunk 7]
```diff
@@ -16,14 +16,14 @@ toc::[levels=1]
...
 
-* link:./templates/sso-cd-https.adoc[sso-cd-https]
-* link:./templates/sso-cd-mysql-persistent.adoc[sso-cd-mysql-persistent]
-* link:./templates/sso-cd-mysql.adoc[sso-cd-mysql]
-* link:./templates/sso-cd-postgresql-persistent.adoc[sso-cd-postgresql-persistent]
-* link:./templates/sso-cd-postgresql.adoc[sso-cd-postgresql]
-* link:./templates/sso-cd-x509-https.adoc[sso-cd-x509-https]
-* link:./templates/sso-cd-x509-mysql-persistent.adoc[sso-cd-x509-mysql-persistent]
-* link:./templates/sso-cd-x509-postgresql-persistent.adoc[sso-cd-x509-postgresql-persistent]
+* link:./templates/sso74-https.adoc[sso74-https]
```
Comment (iankko) [inline docs/README.adoc:21]: @mhajas Just out of curiosity, did you regenerate all the `docs` files by re-running the `generate-template-docs.py` script? Or by editing them manually? The former is the preferred form.

File: gen_openj9_templates.sh (hunk 1 above)
Comment (iankko) [reply gen_openj9_templates.sh]: +1 Maybe we could even have `scripts` subdirectory of the `templates` directory to hold this (and the `generate-template-docs.py` one).

File: gen_openj9_templates.sh (hunk 3 above)
Comment (iankko) [reply gen_openj9_templates.sh]: +1 on the suggestion from Hynek as it's more universal (would work regardless of current RH-SSO version) => no need to keep this updated in the future till it works

File: gen_openj9_templates.sh (hunk 2 above)
Comment (iankko) [reply gen_openj9_templates.sh]: +1 Redirect the output into the `"templates/openj9/$original_template"` You can even move it later to proper name if needed. No need to do all the actions in one step. Rather to keep each step atomic.

Or alternatively pre-generate the new template name name first into `$NEW_TEMPLATE` as part of each for loop iteration, and write it there directly.

File: templates/openj9/sso74-openj9-image-stream.json [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+                            "supports": "sso:7.4",
+                            "version": "latest",
+                            "openshift.io/display-name": "Red Hat Single Sign-On 7.4 on OpenJ9"
+                        },
+                        "referencePolicy": {
+                            "type": "Local"
+                        },
+                        "from": {
+                            "kind": "DockerImage",
+                            "name": "registry.redhat.io/redhat-sso-74/redhat-sso74-openj9-openshift-rhel8:1.0"
```
Comment (iankko) [inline templates/openj9/sso74-openj9-image-stream.json]: There's `redhat` two times. This needs to be written in `Namespace/Repository` form. Namespace is `redhat-sso-7`, repository is `sso74-openj9-openshift-rhel8:1.0` => 

`registry.redhat.io/redhat-sso-7/sso74-openj9-openshift-rhel8:1.0`

File: templates/openj9/sso74-openj9-image-stream.json (hunk 5 above)
Comment (iankko) [reply templates/openj9/sso74-openj9-image-stream.json]: Same as above. Change to RH-SSO 7.4 registry_url/namespace/repository form.

File: templates/openj9/sso74-openj9-mysql-persistent.json [hunk 9]
```diff
@@ -0,0 +1,853 @@
+{
+    "kind": "Template",
+    "apiVersion": "v1",
+    "metadata": {
+        "annotations": {
+            "iconClass" : "icon-sso",
+            "tags" : "sso,keycloak,jboss,hidden",
+            "version": "7.4.0.GA",
+            "openshift.io/display-name": "Red Hat Single Sign-On CD 7.4 on OpenJ9 + MySQL (Persistent with passthrough TLS)",
```
Comment (iankko) [inline templates/openj9/sso74-openj9-mysql-persistent.json]: Drop `CD` here.

File: templates/sso74-https.json [hunk 10]
```diff
@@ -5,19 +5,19 @@
         "annotations": {
             "iconClass" : "icon-sso",
             "tags" : "sso,keycloak,jboss,hidden",
-            "version": "7",
-            "openshift.io/display-name": "Red Hat Single Sign-On CD 7 Tech Preview (Ephemeral with passthrough TLS)",
+            "version": "7.4.0.GA",
```
Comment (iankko) [inline templates/sso74-https.json:8]: `"v7.4.0.GA"` is preferred over `"7.4.0.GA"`. See [current 7.3 tags|https://github.com/jboss-container-images/redhat-sso-7-openshift-image/releases] IIRC the `vRelease` form is used in referenced documentation already, so would be good to be consistent

File: templates/sso74-https.json [hunk 11]
```diff
@@ -5,19 +5,19 @@
...
             "template.openshift.io/support-url": "https://access.redhat.com"
         },
-        "name": "sso-cd-https"
+        "name": "sso74-https"
     },
     "labels": {
-        "template": "sso-cd-https",
-        "rhsso": "7"
+        "template": "sso74-https",
+        "rhsso": "7.4.0.GA"
```
Comment (iankko) [inline templates/sso74-https.json:20]: `s/7.4.0.GA/v7.4.0.GA/g`

File: templates/sso74-image-stream.json [hunk 12]
```diff
@@ -13,165 +13,53 @@
...
-                "name": "redhat-sso-cd-openshift",
+                "name": "redhat-sso74-openshift",
                 "annotations": {
-                    "description": "Red Hat Single Sign-On Continuous Delivery",
-                    "openshift.io/display-name": "Red Hat Single Sign-On Continuous Delivery",
+                    "description": "Red Hat Single Sign-On 7.4 on OpenJDK",
+                    "openshift.io/display-name": "Red Hat Single Sign-On 7.4 on OpenJDK",
                     "openshift.io/provider-display-name": "Red Hat, Inc.",
-                    "version": "6"
+                    "version": "7.4.0.GA"
```
Comment (iankko) [inline templates/sso74-image-stream.json:21]: Same, add `v` prefix for the `version` & `rhsso` values

File: templates/sso74-image-stream.json [hunk 13]
```diff
@@ -13,165 +13,53 @@
...
-                "rhsso": "6"
+                "rhsso": "7.4.0.GA"
             },
             "spec": {
                 "tags": [
                     {
                         "name": "latest",
                         "annotations": {
-                            "description": "Latest build of Red Hat Single Sign-On Continuous Delivery Tech Preview image",
+                            "description": "Latest build of Red Hat Single Sign-On 7.4 OpenJDK image",
```
Comment (iankko) [inline templates/sso74-image-stream.json]: We should be consistent IMHO. If you want / need use `7.4 on OpenJDK` everywhere (in description, display name etc. fields)

File: templates/sso74-image-stream.json [hunk 14]
```diff
@@ -13,165 +13,53 @@
...
-                            "openshift.io/display-name": "Red Hat Single Sign-On Continuous Delivery (Tech Preview)"
+                            "openshift.io/display-name": "Red Hat Single Sign-On 7.4 on OpenJDK"
                         },
                         "referencePolicy": {
                             "type": "Local"
                         },
                         "from": {
                             "kind": "DockerImage",
-                            "name": "registry.redhat.io/redhat-sso-cd-tech-preview/sso-cd-openshift:latest"
+                            "name": "registry.redhat.io/redhat-sso-74/redhat-sso74-openshift-rhel8:1.0"
```
Comment (iankko) [inline templates/sso74-image-stream.json]: Two times `redhat` here. Proper form is `registry.redhat.io/redhat-sso-7/sso74-openshift-rhel8:1.0`

File: templates/sso74-image-stream.json [hunk 15]
```diff
@@ -13,165 +13,53 @@
...
-                    },
-                    {
-                        "name": "7.0",
-                        "referencePolicy": {
-                            "type": "Local"
-                        },
-                        "from": {
-                            "kind": "ImageStreamTag",
-                            "name": "7"
+                            "name": "registry.redhat.io/redhat-sso-7/sso73-openshift:1.0"
```
Comment (iankko) [inline templates/sso74-image-stream.json]: Replace with `registry.redhat.io/redhat-sso-7/sso74-openshift-rhel8:1.0`

File: templates/sso74-mysql-persistent.json [hunk 16]
```diff
@@ -5,19 +5,19 @@
         "annotations": {
             "iconClass" : "icon-sso",
             "tags" : "sso,keycloak,jboss,hidden",
-            "version": "7",
-            "openshift.io/display-name": "Red Hat Single Sign-On CD 7 Tech Preview + MySQL (Persistent with passthrough TLS)",
+            "version": "7.4.0.GA",
```
Comment (iankko) [inline templates/sso74-mysql-persistent.json:8]: Prefix with `v` (both `version` & `rhsso` below)

File: templates/sso74-mysql-persistent.json [hunk 17]
```diff
@@ -5,19 +5,19 @@
...
+            "openshift.io/display-name": "Red Hat Single Sign-On CD 7.4 on OpenJDK + MySQL (Persistent with passthrough TLS)",
```
Comment (iankko) [inline templates/sso74-mysql-persistent.json]: Drop `CD`

File: templates/sso74-mysql.json [hunk 18]
```diff
@@ -5,19 +5,19 @@
         "annotations": {
             "iconClass" : "icon-sso",
             "tags" : "sso,keycloak,jboss,hidden",
-            "version": "7",
-            "openshift.io/display-name": "Red Hat Single Sign-On CD 7 Tech Preview + MySQL (Ephemeral with passthrough TLS)",
+            "version": "7.4.0.GA",
```
Comment (iankko) [inline templates/sso74-mysql.json:8]: Prefix with `v`

File: templates/sso74-postgresql-persistent.json [hunk 19]
```diff
@@ -5,19 +5,19 @@
         "annotations": {
             "iconClass" : "icon-sso",
             "tags" : "sso,keycloak,jboss,hidden",
-            "version": "7",
-            "openshift.io/display-name": "Red Hat Single Sign-On CD 7 TechPreview + PostgreSQL (Persistent with passthrough TLS)",
+            "version": "7.4.0.GA",
```
Comment (iankko) [inline templates/sso74-postgresql-persistent.json:8]: prefix with `v` (both for `version` & `rhsso` value below)

**Review by iankko — COMMENTED** [review]
@mhajas Couple of small changes needed yet, but LGTM in overall

Double check the text for presence of expressions like:
* `Red Hat Single Sign-On CD 7.4` or `RH-SSO CD` (replace the former with `Red Hat Single Sign-On 7.4`, the latter with `RH-SSO`, IOW drop `CD` everywhere)
* Check all text for expressions like `sso-cd` & drop / update it if present

File: templates/sso74-https.json (hunk 11 above)
Comment (hmlnarik) [reply templates/sso74-https.json:20]: @iankko What is the justification for this suggestion?

File: templates/sso74-postgresql-persistent.json (hunk 19 above)
Comment (hmlnarik) [reply templates/sso74-postgresql-persistent.json:8]: @iankko What is the justification for this suggestion?

File: templates/sso74-https.json (hunk 11 above)
Comment (iankko) [reply templates/sso74-https.json:20]: For the very least the name of the tag is used in the errata instructions to update the templates to a specific version for the very least, e.g. [RHEA-2020:0446](https://access.redhat.com/errata/RHEA-2020:0446) reads:

```
...
oc replace -n openshift --force -f \
https://raw.githubusercontent.com/jboss-container-images/redhat-sso-7-openshift-image/v7.3.6.GA/templates/${resource}
...
```

This used to be the tag naming scheme, we used till now. It can be changed, if desired, but in that case we will need to update it on all places previously referencing the `v.*` form

File: templates/sso74-postgresql-persistent.json (hunk 19 above)
Comment (iankko) [reply templates/sso74-postgresql-persistent.json:8]: Same as above. The `v` form is used in the instructions on how to install the updated templates

Comment (hmlnarik) [reply templates/sso74-postgresql-persistent.json:8]: Which instructions? Should they be then updated?
7.3 templates never used `v` form so I'm now confused why we should introduce it now.

File: docs/README.adoc (hunk 7 above)
Comment (mhajas) [reply docs/README.adoc:21]: I used the script

File: templates/sso74-image-stream.json (hunk 14 above)
Comment (mhajas) [reply templates/sso74-image-stream.json]: Looking at 73 branch there is only `registry.redhat.io/redhat-sso-7/sso73-openshift:1.0`, should there be this `-rhel8` suffix? It was in CD releases but not sure it should stay there.

File: templates/sso74-postgresql-persistent.json (hunk 19 above)
Comment (iankko) [reply templates/sso74-postgresql-persistent.json:8]: Ah, right. You are correct. We use the `v` prefix only for tags. Ignore my previous comments

File: docs/README.adoc (hunk 7 above)
Comment (iankko) [reply docs/README.adoc:21]: Ok, great, thanks

File: templates/sso74-https.json (hunk 10 above)
Comment (iankko) [reply templates/sso74-https.json:8]: Not needed (see related discussion), closing.

File: templates/sso74-https.json (hunk 11 above)
Comment (iankko) [reply templates/sso74-https.json:20]: Not needed, confused with the tag notation.

File: templates/sso74-image-stream.json (hunk 12 above)
Comment (iankko) [reply templates/sso74-image-stream.json:21]: Not needed

File: templates/sso74-image-stream.json (hunk 14 above)
Comment (iankko) [reply templates/sso74-image-stream.json]: Yes, the `-rhel8` suffix should be there (it's not present in RH-SSO 7.3 form yet, because it wasn't needed yet). But as the count of base image raises, it's needed to differentiate them.

File: templates/sso74-mysql-persistent.json (hunk 16 above)
Comment (iankko) [reply templates/sso74-mysql-persistent.json:8]: Not needed

File: templates/sso74-mysql.json (hunk 18 above)
Comment (iankko) [reply templates/sso74-mysql.json:8]: Not needed

File: templates/sso74-mysql-persistent.json (hunk 16 above)
Comment (iankko) [reply templates/sso74-mysql-persistent.json:8]: Not needed

File: templates/sso74-mysql.json (hunk 18 above)
Comment (iankko) [reply templates/sso74-mysql.json:8]: Not needed

File: gen_openj9_templates.sh (hunk 1 above)
Comment (mhajas) [reply gen_openj9_templates.sh]: Done, there were some issues with a different working directory, but I fixed them

File: templates/sso74-image-stream.json (hunk 14 above)
Comment (mhajas) [reply templates/sso74-image-stream.json]: Thank you @iankko, I fixed it

**Comment by mhajas** [comment]
Thank you very much for the detailed review @iankko and @hmlnarik! Sorry for the number of issues. I think I fixed all of them, could you please have another look?

File: templates/scripts/gen_template_docs.py [hunk 20]
```diff
@@ -20,15 +20,11 @@
...
+template_dirs = [ 'templates', 'templates/openj9' ]
 amq_ssl_desc = None
 
-LINKS =  { "redhat-sso-cd-openshift:1.0": "../../templates/sso-cd-openshift{outfilesuffix}[`redhat-sso-7-tech-preview/sso-cd-openshift`]",
-           "redhat-sso-cd-openshift:5.0": "../../templates/sso-cd-openshift{outfilesuffix}[`redhat-sso-7-tech-preview/sso-cd-openshift`]",
-           "redhat-sso-cd-openshift:6.0": "../../templates/sso-cd-openshift{outfilesuffix}[`redhat-sso-7-tech-preview/sso-cd-openshift`]",
-           "redhat-sso-cd-openshift:6": "../../templates/sso-cd-openshift{outfilesuffix}[`redhat-sso-7-tech-preview/sso-cd-openshift`]",
-           "redhat-sso-cd-openshift-rhel8:7.0": "../../templates/sso-cd-openshift-rhel8{outfilesuffix}[`redhat-sso-cd-tech-preview/sso-cd-openshift-rhel8`]",
-           "redhat-sso-cd-openshift-rhel8:7": "../../templates/sso-cd-openshift-rhel8{outfilesuffix}[`redhat-sso-cd-tech-preview/sso-cd-openshift-rhel8`]"}
+LINKS =  { "redhat-sso74-openshift-rhel:1.0": "../../templates/sso74-openshift{outfilesuffix}[`redhat-sso-7/sso74-openshift`]",
```
Comment (iankko) [inline templates/scripts/gen_template_docs.py:26]: @mhajas The `LINKS` hash can be removed altogether (doesn't seem to be used anymore) AFAICT. Proof of concept:

```
$ git clone -b KEYCLOAK-12845 git@github.com:mhajas/redhat-sso-7-openshift-image.git
```

```
$ cd redhat-sso-7-openshift-image/templates/scripts/
```

```
$ git diff gen_template_docs.py
diff --git a/templates/scripts/gen_template_docs.py b/templates/scripts/gen_template_docs.py
index 9a1d59b..f758c85 100755
--- a/templates/scripts/gen_template_docs.py
+++ b/templates/scripts/gen_template_docs.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # gen_template_doc.py
 # Kyle Liberti <kliberti@redhat.com>, Jonathan Dowland <jdowland@redhat.com>
 # ver:  Python 3
@@ -23,8 +23,8 @@ TEMPLATE_DOCS = "docs/"
 template_dirs = [ 'templates', 'templates/openj9' ]
 amq_ssl_desc = None
 
-LINKS =  { "redhat-sso74-openshift-rhel:1.0": "../../templates/sso74-openshift{outfilesuffix}[`redhat-sso-7/sso74-openshift`]",
-           "redhat-sso74-openj9-openshift-rhel:1.0": "../../templates/openj9/sso74-openj9-openshift{outfilesuffix}[`redhat-sso-7/sso74-openshift`]"}
+#LINKS =  { "redhat-sso74-openshift-rhel:1.0": "../../templates/sso74-openshift{outfilesuffix}[`redhat-sso-7/sso74-openshift`]",
+#           "redhat-sso74-openj9-openshift-rhel:1.0": "../../templates/openj9/sso74-openj9-openshift{outfilesuffix}[`redhat-sso-7/sso74-openshift`]"}
 
 PARAMETER_VALUES = {"APPLICATION_DOMAIN": "secure-app.test.router.default.local", \
                    "SOURCE_REPOSITORY_URL": "https://github.com/jboss-openshift/openshift-examples.git", \
```

```
$ ./gen_template_docs.py 
Generating docs/templates/sso74-https.adoc...
Generating docs/templates/sso74-mysql-persistent.adoc...
Generating docs/templates/sso74-mysql.adoc...
Generating docs/templates/sso74-postgresql-persistent.adoc...
Generating docs/templates/sso74-postgresql.adoc...
Generating docs/templates/sso74-x509-https.adoc...
Generating docs/templates/sso74-x509-mysql-persistent.adoc...
Generating docs/templates/sso74-x509-postgresql-persistent.adoc...
Generating docs/templates/openj9/sso74-openj9-https.adoc...
Generating docs/templates/openj9/sso74-openj9-mysql-persistent.adoc...
Generating docs/templates/openj9/sso74-openj9-mysql.adoc...
Generating docs/templates/openj9/sso74-openj9-postgresql-persistent.adoc...
Generating docs/templates/openj9/sso74-openj9-postgresql.adoc...
Generating docs/templates/openj9/sso74-openj9-x509-https.adoc...
Generating docs/templates/openj9/sso74-openj9-x509-mysql-persistent.adoc...
Generating docs/templates/openj9/sso74-openj9-x509-postgresql-persistent.adoc...
```

```
$ git status
On branch KEYCLOAK-12845
Your branch is up to date with 'origin/KEYCLOAK-12845'.

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	modified:   ../../docs/templates/openj9/sso74-openj9-https.adoc
	modified:   ../../docs/templates/openj9/sso74-openj9-mysql-persistent.adoc
	modified:   ../../docs/templates/openj9/sso74-openj9-mysql.adoc
	modified:   ../../docs/templates/openj9/sso74-openj9-postgresql-persistent.adoc
	modified:   ../../docs/templates/openj9/sso74-openj9-postgresql.adoc
	modified:   ../../docs/templates/openj9/sso74-openj9-x509-https.adoc
	modified:   ../../docs/templates/openj9/sso74-openj9-x509-mysql-persistent.adoc
	modified:   ../../docs/templates/openj9/sso74-openj9-x509-postgresql-persistent.adoc
	modified:   gen_template_docs.py

no changes added to commit (use "git add" and/or "git commit -a")
```
Just to retest openjdk templates would be generated properly too:
```
[iankko@localhost templates]$ pwd
/tmp/github/redhat-sso-7-openshift-image/templates
```
- Make some demo change (bump image stream tag in PostgreSQL persistent templates to `12`):
```
for template in $(ls | grep postgresql-persistent); do \
    sed -i "/\"name\": \"POSTGRESQL_IMAGE_STREAM_TAG\",/{N;s/\"value\": \"10\",/\"value\": \"12\",/g}" \
              $template; \
done
```

- Verify they got changed:
```
$ git diff *-persistent*
diff --git a/templates/sso74-postgresql-persistent.json b/templates/sso74-postgresql-persistent.json
index 7aa4f14..6e0e718 100644
--- a/templates/sso74-postgresql-persistent.json
+++ b/templates/sso74-postgresql-persistent.json
@@ -256,7 +256,7 @@
             "displayName": "PostgreSQL Image Stream Tag",
             "description": "The tag to use for the \"postgresql\" image stream. Typically, this aligns with the major.minor version of PostgreSQL.",
             "name": "POSTGRESQL_IMAGE_STREAM_TAG",
-            "value": "10",
+            "value": "12",
             "required": true
         },
         {
diff --git a/templates/sso74-x509-postgresql-persistent.json b/templates/sso74-x509-postgresql-persistent.json
index 728fdee..b61518e 100644
--- a/templates/sso74-x509-postgresql-persistent.json
+++ b/templates/sso74-x509-postgresql-persistent.json
@@ -158,7 +158,7 @@
             "displayName": "PostgreSQL Image Stream Tag",
             "description": "The tag to use for the \"postgresql\" image stream.  Typically, this aligns with the major.minor version of PostgreSQL.",
             "name": "POSTGRESQL_IMAGE_STREAM_TAG",
-            "value": "10",
+            "value": "12",
             "required": true
         },
         {
```

- Regenerate the docs:
```
cd scripts/
```

```
$ ./gen_template_docs.py
```

```
$ git diff ../../docs/templates/sso74-x509-postgresql-persistent.adoc
diff --git a/docs/templates/sso74-x509-postgresql-persistent.adoc b/docs/templates/sso74-x509-postgresql-persistent.adoc
index 08cd862..80505d3 100644
--- a/docs/templates/sso74-x509-postgresql-persistent.adoc
+++ b/docs/templates/sso74-x509-postgresql-persistent.adoc
@@ -42,7 +42,7 @@ https://docs.openshift.org/latest/architecture/core_concepts/templates.html#para
 |`SSO_REALM` | `SSO_REALM` | Realm to be created in the RH-SSO server (e.g. demorealm). | `${SSO_REALM}` | False
 |`SSO_SERVICE_USERNAME` | `SSO_SERVICE_USERNAME` | The username used to access the RH-SSO service. This is used by clients to create the appliction client(s) within the specified RH-SSO realm. | `${SSO_SERVICE_USERNAME}` | False
 |`SSO_SERVICE_PASSWORD` | `SSO_SERVICE_PASSWORD` | The password for the RH-SSO service user. | `${SSO_SERVICE_PASSWORD}` | False
-|`POSTGRESQL_IMAGE_STREAM_TAG` | -- | The tag to use for the "postgresql" image stream.  Typically, this aligns with the major.minor version of PostgreSQL. | 10 | True
+|`POSTGRESQL_IMAGE_STREAM_TAG` | -- | The tag to use for the "postgresql" image stream.  Typically, this aligns with the major.minor version of PostgreSQL. | 12 | True
 |`MEMORY_LIMIT` | -- | Container memory limit. | 1Gi | False
 |=======================================================================
```

TL; DR: So AFAICT the `LINKS` hash can be removed completely, as it doesn't seem to be used any more for the RH-SSO images.

@jmtd Hey Jon, do you know what is / was the purpose of the `LINKS` associative array in the `gen_template_docs.py` script / what it was used for?

File: templates/openj9/sso74-openj9-https.json [hunk 21]
```diff
@@ -0,0 +1,579 @@
+{
+    "kind": "Template",
+    "apiVersion": "v1",
+    "metadata": {
+        "annotations": {
+            "iconClass" : "icon-sso",
+            "tags" : "sso,keycloak,jboss,hidden",
+            "version": "v7.4.0.GA",
```
Comment (iankko) [inline templates/openj9/sso74-openj9-https.json]: @mhajas Please undo the `version` & `rhsso` `s/7.4.0.GA/v7.4.0.GA/g` replacements. @hmlnarik is right, this was never used in RH-SSO 7.3 yet. I confused it yesterday with the tag name format (too many issues to handle yesterday at once, so overlooked this).

Sorry for the confusion (additional work). Thought I closed all the wrong comments yesterday already, but it was obviously too late

**Review by hmlnarik — APPROVED** [review]

**Review by iankko — APPROVED** [review]
LGTM Thanks, @mhajas !
